cmd/login/runner.go

```diff
@@ -181,12 +181,22 @@ func (r *runner) loginWithKubeContextName(ctx context.Context, contextName strin
 // one with the name derived from the installation code name.
 func (r *runner) loginWithCodeName(ctx context.Context, codeName string) error {
 	var contextAlreadySelected bool
+	var newLoginRequired bool
+	var selfContained bool
 
 	contextName := kubeconfig.GenerateKubeContextName(codeName)
-	err := switchContext(ctx, r.k8sConfigAccess, contextName)
+	err := switchContext(ctx, r.k8sConfigAccess, contextName, r.flag.KeepContext || (len(r.flag.SelfContained) > 0))
 	if IsContextAlreadySelected(err) {
 		contextAlreadySelected = true
 	} else if IsNewLoginRequired(err) {
+		newLoginRequired = true
```

Comment:
I noticed while testing that this condition isn't triggered, so when I try to login using a code name with the self-contained flag (`kubectl gs login <codename> --self-contained`), I get a "Token renewal failed" error because there's no refresh token in the empty file.

I think `switchContext()` isn't returning a `newLoginRequiredError` though it does return `tokenRenewalFailedError` when renewing the token fails?

Reply:
I tried this and it happened to me, too. Do you think it would be safe to continue with the new login despite the error in case we are requesting a self contained kubeconfig?

Reply:
I think that makes sense 👍

Reply:
Let me know if you still see the issue.

Reply:
Not anymore - looks good! Thanks for checking this out 😄